components/nexusoperations/completion.go

```diff
@@ -75,18 +75,19 @@ func handleUnsuccessfulOperationError(
 	if err != nil {
 		return hsm.TransitionOutput{}, err
 	}
+	failure, err := commonnexus.UnsuccessfulOperationErrorToTemporalFailure(opFailedError)
+	if err != nil {
+		return hsm.TransitionOutput{}, err
+	}
```

Comment:
Does it make sense that you're overwriting the operation failure with an error during the conversion of the failure?
What's gonna happen to the operation when you return an error here?

Reply:
I don't expect this to ever fail TBH but I'm not sure that I understand the concern.

Reply:
Say the operation failed, and you failed to parse the operation failure here and returned an error that is related to failing to parse the operation error.
So, the operation state machine won't know that the operation actually failed.

Reply:
If you can't record the failure, it's okay to just drop the result IMHO. Also, again, not expecting this to ever fail.